apiserver/facades/agent/uniter/status.go

```diff
@@ -22,13 +23,13 @@ type StatusAPI struct {
 }
 
 // NewStatusAPI creates a new server-side Status setter API facade.
-func NewStatusAPI(st *state.State, getCanModify common.GetAuthFunc) *StatusAPI {
+func NewStatusAPI(st *state.State, getCanModify common.GetAuthFunc, leadershipChecker leadership.Checker) *StatusAPI {
 	// TODO(fwereade): so *all* of these have exactly the same auth
 	// characteristics? I think not.
 	unitSetter := common.NewStatusSetter(st, getCanModify)
 	unitGetter := common.NewStatusGetter(st, getCanModify)
-	serviceSetter := common.NewServiceStatusSetter(st, getCanModify)
-	serviceGetter := common.NewApplicationStatusGetter(st, getCanModify)
+	serviceSetter := common.NewApplicationStatusSetter(st, getCanModify, leadershipChecker)
```

Comment:
I am wondering why there are 2?

Reply:
there's getter/setter for units and applications
the diff is a bit confusing